packages/js/client/src/__tests__/core/uri-resolution/uri-resolution.spec.ts

```diff
@@ -0,0 +1,440 @@
+import { Uri } from "../../../../";
+import { buildWrapper } from "@polywrap/test-env-js";
+import { GetPathToTestWrappers } from "@polywrap/test-cases";
+import {
+  coreInterfaceUris,
+  IUriResolutionStep,
+  PluginModule,
+  UriPackageOrWrapper,
+  UriResolutionContext,
+  UriResolutionResult,
+} from "@polywrap/core-js";
+import { buildCleanUriHistory, getUriResolutionPath } from "@polywrap/uri-resolvers-js";
+import { getClient } from "../../utils/getClient";
+import fs from "fs";
+import { WrapManifest } from "@polywrap/wrap-manifest-types-js";
+import { Result } from "@polywrap/result";
+
+jest.setTimeout(200000);
+
+const wrapperPath = `${GetPathToTestWrappers()}/wasm-as/simple`;
+const wrapperUri = new Uri(`wrap://file/${wrapperPath}/build`);
+
+const simpleFsResolverWrapperPath = `${GetPathToTestWrappers()}/wasm-as/simple-fs-resolver`;
+const simpleFsResolverWrapperUri = new Uri(
+  `wrap://file/${simpleFsResolverWrapperPath}/build`
+);
+
+const simpleRedirectResolverWrapperPath = `${GetPathToTestWrappers()}/wasm-as/simple-redirect-resolver`;
+const simpleRedirectResolverWrapperUri = new Uri(
+  `wrap://file/${simpleRedirectResolverWrapperPath}/build`
+);
+
+const expectResultWithHistory = async (
+  receivedResult: Result<UriPackageOrWrapper, unknown>,
+  expectedResult: Result<UriPackageOrWrapper, unknown>,
+  uriHistory: IUriResolutionStep<unknown>[],
+  historyFileName: string
+): Promise<void> => {
+  if (historyFileName && uriHistory) {
+    await expectHistory(uriHistory, historyFileName);
+  }
+  
+  expect(receivedResult).toEqual(expectedResult);
+};
+
+const expectHistory = async (
+  receivedHistory: IUriResolutionStep<unknown>[] | undefined,
+  historyFileName: string
+): Promise<void> => {
+  if (!receivedHistory) {
+    fail("History is not defined");
+  }
+
+  let expectedCleanHistory = await fs.promises.readFile(
+    `${__dirname}/histories/${historyFileName}.json`,
+    "utf-8"
+  );
+
+  const receivedCleanHistory = replaceAll(
+    JSON.stringify(buildCleanUriHistory(receivedHistory), null, 2),
+    `${GetPathToTestWrappers()}/wasm-as`,
+    "$root-wrapper-dir"
+  );
+
+  expect(receivedCleanHistory).toEqual(
+    JSON.stringify(JSON.parse(expectedCleanHistory), null, 2)
+  );
+};
+
+const expectPackageWithHistory = async (
```

Comment:
This function is not being used anywhere, do we need it?

Reply:
Not anymore, yes in the future, but I'll remove it for now